src/modules/services/postgres.nix

```diff
@@ -276,6 +281,19 @@ in
         CREATE ROLE bar;
       '';
     };
+
+    hbaConf = lib.mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = ''
+        The path to a custom pg_hba.conf file to copy into the postgres installation. This
+        allows for custom connection rules that you want to establish on the server. If
+        you're using flakes remember to add the file to version control!
+      '';
+      example = lib.literalExpression ''
+        "${./my-custom/directory/to/pg_hba.conf}"
```

Comment:
Needs to use `builtins.readFile`

Reply:
Isn't just putting the file path enough to copy it into the Nix store? From what I understand it copies the file into a path in the nix store and then just references it.

Source: https://stackoverflow.com/questions/43850371/when-does-a-nix-path-type-make-it-into-the-nix-store-when-converted-to-a-string

Reply:
Yeah but it expect it a string now, so you don't want the contents of pg_hba.conf to be a nix store path, but the contents of it.

Reply:
Gotcha, ok! I'll do the changes in a minute